cmake/VecCoreConfig.cmake.in

```diff
@@ -17,12 +17,9 @@ list(APPEND CMAKE_MODULE_PATH "${VecCore_CMAKE_DIR}")
 include(CMakeFindDependencyMacro)
 
 if (VecCore_FIND_COMPONENTS MATCHES "CUDA")
-  find_dependency(CUDA 7.5)
-  if(CUDA_FOUND)
     set(VecCore_CUDA_FOUND True)
     set(VecCore_CUDA_DEFINITIONS -DVECCORE_ENABLE_CUDA)
-    set(VecCore_CUDA_INCLUDE_DIR ${CUDA_INCLUDE_DIRS})
-  endif()
+    set(VecCore_CUDA_INCLUDE_DIR ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
```

Comment:
This cannot go as is, otherwise dependent projects will break when VecCore's CUDA component is required in, say `find_package(VecCore REQUIRED COMPONENTS CUDA)`.

Reply:
In this case I am not sure how to proceed.  This change seems to be working fine, as tested from the VecGeom side. Namely, VecGeom build prints the CMP0146 warning when pointing to VecCore 0.8.1, while to warning is printed when pointing to VecCore as changed in this pull request.

Reply:
Ok, I will test the changes tomorrow. Maybe with CUDA being promoted to have native support we can actually drop the `find_package` component support for CUDA (as the top project already sets that up, like VecGeom). I think VecCore is starting to show its age a bit.

Reply:
I don't know if this information is useful for you, but I've made some further testing with VecGeom and Celeritas:

1) against VecCore 0.8.1:
  - CMP0146 warning is printed, but otherwise builds and runs ok.
  - adding cmake_policy(SET CMP0146 NEW) near the top of VecGeom's main `CMakeLists.txt` file, then VecGeom's CMake reconfiguration fails with this error:
```
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
CMake Error at /srv/software/el8/x86_64/hpc/cmake/3.27.3/share/cmake-3.27/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  By not providing "FindCUDA.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "CUDA", but
  CMake did not find one.

  Could not find a package configuration file provided by "CUDA" (requested
  version 7.5) with any of the following names:

    CUDAConfig.cmake
    cuda-config.cmake

  Add the installation prefix of "CUDA" to CMAKE_PREFIX_PATH or set
  "CUDA_DIR" to a directory containing one of the above files.  If "CUDA"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /work1/celeritas/lima/work/local/gcc-12.3.0/veccore/0.8.1/lib64/cmake/VecCore/VecCoreConfig.cmake:44 (find_dependency)
  CMakeLists.txt:224 (find_package)


-- Configuring incomplete, errors occurred!
```

2) against VecCore with the changes in this PR:
   No CMP0146 warnings are printed, whether or not CMP0146 is set to NEW.
   VecGeom builds fine with CUDA enabled, and all tests passed.
   I also built Celelitas against this build of VecGeom (+VecCore), and also all tests passed.

Reply:
<img width="992" alt="Screenshot 2024-01-30 at 1 01 25 PM" src="https://github.com/root-project/veccore/assets/6361464/b30812ba-b248-4713-b33c-3e799f900861">